blaze/compute/numba.py

```diff
@@ -196,15 +198,20 @@ def _get_numba_ufunc(expr):
     else:
         leaves = expr._leaves()
 
-    # we may not have a Broadcast instance because arithmetic expressions can
-    # be vectorized so we use getattr
     s, scope = funcstr(leaves, expr)
 
     scope = dict((k, numba.jit(nopython=True)(v) if callable(v) else v)
                  for k, v in scope.items())
+    # get the func
     func = eval(s, scope)
+    # get the signature
     sig = compute_signature(expr)
-    return numba.vectorize([sig], nopython=True)(func)
+    # vectorize is currently not thread safe. So lock the thread.
+    # TODO FIXME remove this when numba has made vectorize thread safe.
+    lock.acquire()
+    ufunc = numba.vectorize([sig], nopython=True)(func)
+    lock.release()
```

Comment:
Just PSA (not a request to change anything) but are you aware that locks function as context managers?

```python
with lock:
    ufunc = ...
```